src/core/src/lib/extensions/core/core.ts

```diff
@@ -29,6 +29,32 @@ export class CoreExtension implements FormlyExtension {
       });
     }
 
+    Object.defineProperty(field, 'get', {
+      value: (key) => {
+        let formlyField: FormlyFieldConfig = null;
+
+        const findField = (f: FormlyFieldConfig, keyValue: string | number) => {
+          if (f.fieldGroup && f.fieldGroup.length > 0) {
+            const match = f.fieldGroup.find((fg) => fg.key && fg.key === keyValue);
```

Comment:
we need to manage two extra cases:
- nested key `address.city`.
- finding field by `id` which is useful for a field that doesn't have a key